easybuild/easyblocks/i/imkl.py

```diff
@@ -77,6 +77,34 @@ def prepare_step(self, *args, **kwargs):
             super(EB_imkl, self).prepare_step(*args, **kwargs)
         else:
             super(EB_imkl, self).prepare_step(*args, **kwargs)
+        if self.cfg['interfaces']:
+            cdftlibs = ['fftw2x_cdft']
+            if LooseVersion(self.version) >= LooseVersion('10.3'):
+                cdftlibs.append('fftw3x_cdft')
+            mpi_spec = None
+            # check whether MPI_FAMILY constant is defined, so mpi_family() can be used
+            if hasattr(self.toolchain, 'MPI_FAMILY') and self.toolchain.MPI_FAMILY is not None:
+                mpi_spec_by_fam = {
+                    toolchain.MPICH: 'mpich2',  # MPICH is MPICH v3.x, which is MPICH2 compatible
+                    toolchain.MPICH2: 'mpich2',
+                    toolchain.MVAPICH2: 'mpich2',
+                    toolchain.OPENMPI: 'openmpi',
+                }
+                mpi_fam = self.toolchain.mpi_family()
+                mpi_spec = mpi_spec_by_fam.get(mpi_fam)
+                self.log.debug("Determined MPI specification based on MPI toolchain component: %s" % mpi_spec)
+            else:
+                # can't use toolchain.mpi_family, because of dummy toolchain
+                if get_software_root('MPICH2') or get_software_root('MVAPICH2'):
+                    mpi_spec = 'mpich2'
+                elif get_software_root('OpenMPI'):
+                    mpi_spec = 'openmpi'
+                elif not get_software_root('impi'):
+                    # no compatible MPI found: do not build cdft
+                    cdftlibs = []
+                self.log.debug("Determined MPI specification based on loaded MPI module: %s" % mpi_spec)
+            self.cdftlibs = cdftlibs
+            self.mpi_spec = mpi_spec
```

Comment:
@bartoldeman Please initialize both of these in the constructor of the easyblock (`__init__`)